bdb/file.c

```diff
@@ -2196,9 +2196,12 @@ static void bdb_appsock(netinfo_type *netinfo, SBUF2 *sb)
 }
 
 extern int gbl_pstack_self;
+static int running_pstack = 0;
 void pstack_self(void)
 {
-    if (!gbl_pstack_self) return;
+    if (!gbl_pstack_self || running_pstack)
```

Comment:
since this is to prevent concurrent pstack callers, does this check need to be protected by a mutex?

Reply:
> New version protects this under a mutex